File: web/src/types/exercise.ts
// Exercise-related types

export type MuscleGroup =
  | 'chest'
  | 'back'
  | 'shoulders'
  | 'biceps'
  | 'triceps'
  | 'forearms'
  | 'abs'
  | 'obliques'
  | 'quads'
  | 'hamstrings'
  | 'glutes'
  | 'calves'
  | 'traps'
  | 'lats';

export interface Exercise {
  id: string;
  name: string;
  category?: 'compound' | 'isolation' | 'cardio'; // Optional - not always set from backend
  muscleGroups?: string[]; // Optional - array of muscle group names
  equipment?: string | null; // Single equipment name or null
  instructions?: string[]; // Optional - not stored in backend yet
  bodyweight?: boolean; // True for exercises like pullups, pushups, dips
}

File: backend/app/schemas/exercise.py
from pydantic import BaseModel, Field
from typing import List, Literal, Optional


class ExerciseBase(BaseModel):
    name: str
    category: Literal["compound", "isolation"]
    muscleGroups: List[str]
    equipment: List[str]
    instructions: str


class ExerciseCreate(ExerciseBase):
    pass


class ExerciseUpdate(BaseModel):
    name: Optional[str] = None
    category: Optional[Literal["compound", "isolation"]] = None
    muscleGroups: Optional[List[str]] = None
    equipment: Optional[List[str]] = None
    instructions: Optional[str] = None


class Exercise(ExerciseBase):
    id: str

    model_config = {"from_attributes": True}
